### DISCUSSION TITLE
Checkerboard artifacts in reconstructed image

### DISCUSSION THREAD

**Discussion by parthnatekar** [tensorflow/compression#109] [General]
Hi, I'm training an mselpips model on a custom dataset.

I'm getting checkerboard artifacts in my reconstructed image, shown below. This is a part of the background and should hence be completely zero
![Screenshot from 2021-11-05 17-11-55](https://user-images.githubusercontent.com/33252453/140591014-722ae69c-27ac-4a96-9816-5ca66f888f5a.png)

Why does this happen? Do I need to change my hyperparameters?

Thanks

**Comment by fab-jul** [comment]
Hi, not sure what's happening, but note that in the paper, we report that without a GAN loss (which I think is what the `mselpips` model is), we also see gridding. See FigA2 in the appendix here: https://arxiv.org/pdf/2006.09965.pdf

**Reply by parthnatekar** [reply]
Hi, thanks for the reply.

A follow up on this - in archs.py, the final layer activation for the decoder is 'linear', which is what Keras defaults to. However, the MSE and LPIPS losses are designed for 0-1 float outputs (or 0-255 ints). Is this intended?
